### PR TITLE
Coot15, Do not use the transform function if we are using bandicoot in softmax layer

### PR DESCRIPTION
**Pull request by shrit**
@MarkFischinger please feel free to open a PR to replace the current implementation with Padé

### CODE REVIEW

File: src/mlpack/methods/ann/layer/log_softmax_impl.hpp [hunk 1]
```diff
@@ -97,6 +99,22 @@ void LogSoftMaxType<MatType>::Forward(const MatType& input, MatType& output)
...
+
+template<typename MatType>
+void LogSoftMaxType<MatType>::Forward(const MatType& input, MatType& output,
+                                      const typename std::enable_if_t<
+                                      coot::is_coot_type<MatType>::value>*)
+{
+  MatType maxInput = repmat(max(input), input.n_rows, 1);
+  output = (maxInput - input);
+  output = exp(output * -1);
+  maxInput.each_row() += log(sum(output));
```
Comment (shrit) [inline src/mlpack/methods/ann/layer/log_softmax_impl.hpp:112]: technically, we still do not have an `each_row()` function in bandicoot

Comment (rcurtin) [reply src/mlpack/methods/ann/layer/log_softmax_impl.hpp:112]: Yes, it is possible and on my todo list though.  (Soon enough... :smile:)

File: src/mlpack/methods/ann/layer/log_softmax_impl.hpp [hunk 2]
```diff
@@ -64,6 +64,14 @@ LogSoftMaxType<MatType>::operator=(LogSoftMaxType&& other)
...
 template<typename MatType>
 void LogSoftMaxType<MatType>::Forward(const MatType& input, MatType& output)
+{
+  ForwardImpl(input, output);
+}
+
+template<typename MatType>
+void LogSoftMaxType<MatType>::ForwardImpl(const MatType& input, MatType& output,
+                                          const typename std::enable_if_t<
+                                          arma::is_arma_type<MatType>::value>*)
```
Comment (rcurtin) [inline src/mlpack/methods/ann/layer/log_softmax_impl.hpp:74]:
```suggestion
void LogSoftMaxType<MatType>::ForwardImpl(const MatType& input,
                                          MatType& output,
                                          const typename std::enable_if_t<
                                              arma::is_arma_type<MatType>::value>*)
```
The usual style suggestions :smile: sorry if they can be frustrating at times, I'm just hoping to make things look consistent throughout.

**Review by rcurtin — APPROVED** [review]
Looks good to me, although we won't really be able to test it until everything else is in place.

**Pull request merged**